packages/records/lib/db/migrations/20241023085055_create_record_counts_table.ts

```diff
@@ -0,0 +1,31 @@
+import type { Knex } from 'knex';
+
+const TABLE = 'record_counts';
+
+export async function up(knex: Knex): Promise<void> {
+    await knex.transaction(async (trx) => {
+        // TABLE
+        await trx.raw(`
+            CREATE TABLE "${TABLE}" (
+                connection_id integer NOT NULL,
+                environment_id integer NOT NULL,
+                model character varying(255) NOT NULL,
+                object_count integer NOT NULL,
+                updated_at timestamp with time zone NOT NULL DEFAULT CURRENT_TIMESTAMP
+            )
+        `);
+
+        // INDEXES
+        await knex.schema.alterTable(TABLE, function (table) {
+            // Maybe include environment_id on uniqueness and compound index?
```

Comment:
yes for both indexes you need the environment id otherwise there is a risk of collision and index will not be used for querying

Reply:
What exactly is the environment? Based on the schema it looks like it corresponds to a customer's environment?

Reply:
yes, 
account = organisation 
environment is like a workspace inside an or, it contains the scripts, the records, the logs, etc.
Everything should be grouped at the environment level for security/isolation reason. Because two envs can have the  same connection_id, scripts names, etc.